File: api/schemas/fundamentals.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any

class CompanyProfile(BaseModel):
    ticker: str
    company_name: Optional[str] = None
    industry: Optional[str] = None
    sector: Optional[str] = None
    description: Optional[str] = None
    market_cap: Optional[float] = None
    source: str

class KeyMetrics(BaseModel):
    ticker: str
    pe_ratio: Optional[float] = None
    pb_ratio: Optional[float] = None
    roe: Optional[float] = None
    eps: Optional[float] = None
    source: str

class FinancialsResponse(BaseModel):
    ticker: str
    period: str
    data: List[Dict[str, Any]]

File: apps/dsa-web/src/api/fundamentals.ts
import { apiClient } from './client';

export interface CompanyProfile {
  symbol: string;
  name: string;
  sector: string;
  industry: string;
  description: string;
  website: string;
  employees: number;
  market_cap: number;
  pe_ratio: number;
  dividend_yield: number;
}

export interface FinancialMetric {
  year: number;
  revenue: number;
  net_income: number;
  eps: number;
  operating_margin: number;
  fcf: number; // Free cash flow
}

export const fundamentalsApi = {
  getProfile: async (ticker: string): Promise<CompanyProfile> => {
    const { data } = await apiClient.get<CompanyProfile>(`/fundamentals/profile?ticker=${ticker}`);
    return data;
  },

  getFinancials: async (ticker: string, type: 'annual' | 'quarterly' = 'annual'): Promise<FinancialMetric[]> => {
    const { data } = await apiClient.get<FinancialMetric[]>(`/fundamentals/financials?ticker=${ticker}&type=${type}`);
    return data;
  },
  
  getMetrics: async (ticker: string): Promise<Record<string, number>> => {
    const { data } = await apiClient.get<Record<string, number>>(`/fundamentals/metrics?ticker=${ticker}`);
    return data;
  }
};
